src/main/java/org/candlepin/model/AnonymousCloudConsumer.java

```diff
@@ -0,0 +1,271 @@
+ * content access until it is claimed by an owner.
+ */
+@Entity
+@Table(name = AnonymousCloudConsumer.DB_TABLE)
+public class AnonymousCloudConsumer extends AbstractHibernateObject<AnonymousCloudConsumer> {
+
+    /** Name of the table backing this object in the database */
+    public static final String DB_TABLE = "cp_anonymous_cloud_consumers";
+
+    /** Max length for a value in the ID field */
+    public static final int ID_MAX_LENGTH = 32;
+    /** Max length for a value in the UUID field */
+    public static final int UUID_MAX_LENGTH = 36;
+    /** Max length for a value in the cloud account ID field */
+    public static final int CLOUD_ACCOUNT_ID_MAX_LENGTH = 255;
+    /** Max length for a value in the cloud instance ID field */
+    public static final int CLOUD_INSTANCE_ID_MAX_LENGTH = 85;
+    /** Max length for a value in the cloud provider short name field */
+    public static final int CLOUD_PROVIDER_SHORT_NAME_MAX_LENGTH = 15;
+    /** Max length for a value in the product ID field */
+    public static final int PRODUCT_ID_MAX_LENGTH = 32;
+
+    @Id
+    @GeneratedValue(generator = "system-uuid")
+    @GenericGenerator(name = "system-uuid", strategy = "uuid")
+    @NotNull
+    private String id;
+
+    @Column(unique = true)
+    @NotNull
+    private String uuid;
+
+    @Column(name = "cloud_account_id")
+    @NotNull
+    private String cloudAccountId;
+
+    @Column(name = "cloud_instance_id")
+    @NotNull
+    private String cloudInstanceId;
+
+    @Column(name = "cloud_provider_short_name")
+    @NotNull
+    private String cloudProviderShortName;
+
+    @Column(name = "product_id")
+    @NotNull
+    private String productId;
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getId() {
+        return this.id;
+    }
+
+    /**
+     * @param id
+     *     the database ID
+     *
+     * @return a reference to this AnonymousCloudConsumer instance
+     */
+    public AnonymousCloudConsumer setId(String id) {
+        if (id == null) {
+            throw new IllegalArgumentException("id is null");
+        }
+
+        if (id.length() > ID_MAX_LENGTH) {
+            throw new IllegalArgumentException("id exceeds the max length");
+        }
+
+        this.id = id;
+        return this;
+    }
+
+    /**
+     * @return the UUID for this AnonymousCloudConsumer
+     */
+    public String getUuid() {
+        return this.uuid;
+    }
+
+    /**
+     * @param uuid
+     *     the UUID to set
+     *
+     * @return a reference to this AnonymousCloudConsumer instance
+     */
+    public AnonymousCloudConsumer setUuid(String uuid) {
+        if (uuid == null) {
+            throw new IllegalArgumentException("uuid is null");
+        }
+
+        if (uuid.length() > UUID_MAX_LENGTH) {
+            throw new IllegalArgumentException("uuid exceeds the max length");
+        }
+
+        this.uuid = uuid;
+        return this;
+    }
+
+    @PrePersist
+    public void ensureUUID() {
```

Comment:
Removed the mutators for UUID and the pre-persist hook, and moved the generation of the UUID to the constructor!

I can create a separate PR to make the same changes for Consumer. It will be much easier to review them in a separate PR!

Reply:
Discussed offline, but we decided to keep the UUID mutators and the validation check in the pre-persist hook.

Reply:
Our wires got crossed somewhere along the line. My intent was to:
- generate a UUID at construction time
- keep the setter with its validators (adding the isBlank check)
- remove the pre-persist hook for UUID generation

Reply:
That is my bad, I misunderstood your message! I have made the changes